platforms/avr/gpio.h

```diff
@@ -25,7 +25,9 @@ typedef uint8_t pin_t;
 #define setPinInput(pin) (DDRx_ADDRESS(pin) &= ~_BV((pin)&0xF), PORTx_ADDRESS(pin) &= ~_BV((pin)&0xF))
 #define setPinInputHigh(pin) (DDRx_ADDRESS(pin) &= ~_BV((pin)&0xF), PORTx_ADDRESS(pin) |= _BV((pin)&0xF))
 #define setPinInputLow(pin) _Static_assert(0, "AVR processors cannot implement an input as pull low")
-#define setPinOutput(pin) (DDRx_ADDRESS(pin) |= _BV((pin)&0xF))
+#define setPinOutputPushPull(pin) (DDRx_ADDRESS(pin) |= _BV((pin)&0xF))
+#define setPinOutputOpenDrain(pin) _Static_assert(0, "AVR platform does not implement an open-drain output")
```

Comment:
Technically it is possible to implement an open-drain output on AVR — set the PORTx bit to 0, then write the inverted output value to the DDRx bit. The problem is that it requires either some conditional code in the `writePin()` implementation, or a separate set of macros like `writeOpenDrainPin()`, `writeOpenDrainPinHigh()`, `writeOpenDrainPinLow()` (with bad consequences if you use a wrong kind of macro for the chosen pin mode).

Reply:
Yep, but nobody has asked for it, and nobody's using it. Figured it'd be a problem for another day.